### PR TITLE
UCS: add ucs_print()

### PR DESCRIPTION
**Pull request by alex-mikheev**
@yosefe please take a look

### CODE REVIEW

**Comment by alex-mikheev** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4309/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4311/ for details (Mellanox internal link).

File: src/ucs/config/global_opts.c [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 ucs_global_opts_t ucs_global_opts = {
     .log_level             = UCS_LOG_LEVEL_WARN,
+    .log_prints            = 0,
```
Comment (yosefe) [inline src/ucs/config/global_opts.c]: maybe "log_print_enable"?

File: src/ucs/config/global_opts.c [hunk 2]
```diff
@@ -73,6 +74,10 @@ static ucs_config_field_t ucs_global_opts_table[] = {
   "How much packet payload to print, at most, in data mode.",
   ucs_offsetof(ucs_global_opts_t, log_data_size), UCS_CONFIG_TYPE_ULONG},
 
+ {"LOG_ENABLE_PRINTS", "n",
```
Comment (yosefe) [inline src/ucs/config/global_opts.c]: LOG_PRINT_ENABLE

File: src/ucs/config/global_opts.c [hunk 3]
```diff
@@ -73,6 +74,10 @@ static ucs_config_field_t ucs_global_opts_table[] = {
...
+  "Enable output of ucs_print(). This option is intended for use by the library evelopers.\n",
```
Comment (yosefe) [inline src/ucs/config/global_opts.c]: library developers

File: src/ucs/debug/log.h [hunk 4]
```diff
@@ -78,6 +78,27 @@
...
+ * to see messages only from specific processes. For example, one may
+ * want to see prints only from rank 0 when debugging MPI.
+ *
+ * The function is intended for debugging only. It should not be used
+ * in the real code.
+ */
+
+#define ucs_print(_message, ...) \
+    do { \
+        if (ucs_global_opts.log_prints) { \
```
Comment (yosefe) [inline src/ucs/debug/log.h]: maybe put this check inside ucs_log_enabled() ?

File: src/ucs/debug/log.h [hunk 5]
```diff
@@ -78,6 +78,27 @@
...
+ * Print a message regardless of current log level. Output can be
+ * enabled/disabled via environment variable/configuration settings.
+ *
+ * During debugging it can be useful to add a few prints to the code
+ * without changing a current log level. Also it is useful to be able
+ * to see messages only from specific processes. For example, one may
+ * want to see prints only from rank 0 when debugging MPI.
+ *
+ * The function is intended for debugging only. It should not be used
+ * in the real code.
```
Comment (yosefe) [inline src/ucs/debug/log.h]: maybe test_jenkins.sh can test this somehow?

**Comment by yosefe** [comment]
bot;mlx:retest

File: src/ucs/debug/log.h (hunk 4 above)
Comment (alex-mikheev) [inline src/ucs/debug/log.h]: I did not want to add another condition check to the fast path

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4317/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:retest

**Comment by yosefe** [comment]
bot:ornl:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4334/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4336/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:ornl:retest

**Pull request merged**